### PR TITLE
Simplify defining custom elements

### PR DESCRIPTION
**Pull request by mfherbst**

### CODE REVIEW

File: examples/custom_potential.jl [hunk 1]
```diff
@@ -7,44 +7,37 @@
...
 
-# We need to extend two methods to access the real and Fourier forms of
-# the potential during the computations performed by DFTK
-function DFTK.local_potential_fourier(el::ElementCustomPotential, q::Real)
-    return el.pot_fourier(q)
+# We extend the two methods providing access to the real and Fourier
+# representation of the potential to DFTK.
+
+function DFTK.local_potential_fourier(el::ElementGaussian, q::Real)
+    el.α * exp(- (q * el.L)^2 / 4)
```
Comment (antoine-levitt) [inline examples/custom_potential.jl:21]: Normalizations look fishy here, there's no pi?

Comment (mfherbst) [reply examples/custom_potential.jl:21]: Yes I think they are, but I copied from Gaspard ...

Comment (mfherbst) [reply examples/custom_potential.jl:21]: Let me take a look.

File: examples/custom_potential.jl [hunk 2]
```diff
@@ -7,44 +7,37 @@
...
-# We set the potential in its real and Fourier forms
-pot_real(x) = exp(-(x/L)^2)
-pot_fourier(q::T) where {T <: Real} = exp(- (q*L)^2 / 4);
-# And finally we build the elements and set their positions in the `atoms`
-# array. Note that in this example `pot_real` is not required as all applications
-# of local potentials are done in the Fourier space.
-nucleus = ElementCustomPotential(pot_real, pot_fourier)
-atoms = [nucleus => [x1*[1,0,0], x2*[1,0,0]]];
+x2 = 0.8
+nucleus = ElementGaussian(1.0, 0.5)
```
Comment (antoine-levitt) [inline examples/custom_potential.jl:37]: might as well give them a negative potential to get bound states

Comment (mfherbst) [reply examples/custom_potential.jl:37]: For GPE?

File: examples/custom_potential.jl (hunk 1 above)
Comment (antoine-levitt) [reply examples/custom_potential.jl:21]: The convention we use is V(q) = int_{R^3} V(x) e^{-iqx} dx (documented in elements.jl) so very likely there's a factor missing

Comment (antoine-levitt) [reply examples/custom_potential.jl:21]: @gkemlin :-)

File: examples/custom_potential.jl (hunk 2 above)
Comment (antoine-levitt) [reply examples/custom_potential.jl:37]: Yeah, sure, this GPE is not very different from the linear model. The current examples gets scattering states (which is fine in a finite box), but would be easier to see with bound states.

**Pull request merged**